presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -280,6 +281,16 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         return columns.build();
     }
 
+    @Override
+    public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint constraint)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, handle.getSchemaTableName());
+        PartitionTable partitionTable = new PartitionTable(handle, typeManager, icebergTable);
+        partitionTable.getTableMetadata();
```

Comment:
Leftover?

Reply:
No, AIUI this is necessary in order to populate the partition table.

Reply:
In the commit I just force-pushed, I changed PartitionTable so that the ConnectorTableMetadata was created in the constructor.  That eliminated the strange API, and allowed a bunch of data members to be final.